### PR TITLE
docs: update DocSearch showcase

### PR DESCRIPTION
**Pull request by shortcuts**
## Pre-flight checklist

- [x] I have read the [Contributing Guidelines on pull requests](https://github.com/facebook/docusaurus/blob/main/CONTRIBUTING.md#pull-requests).
- [ ] **If this is a code change**: I have written unit tests and/or added dogfooding pages to fully verify the new behavior.
- [ ] **If this is a new API or substantial change**: the PR has an accompanying issue (closes #0000) and the maintainers have approved on my working plan.

## Motivation

The link to the DocSearch website repository has changed.

### CODE REVIEW

File: website/src/data/users.tsx [hunk 1]
```diff
@@ -95,12 +95,12 @@ const Users: User[] = [
...
   },
   {
-    title: 'Algolia Docsearch',
+    title: 'Algolia DocSearch',
     description:
       'The best search experience for docs, integrates in minutes, for free',
     preview: require('./showcase/algolia.png'),
     website: 'https://docsearch.algolia.com/',
-    source: 'https://github.com/algolia/docsearch-website',
+    source: 'https://github.com/algolia/docsearch',
```
Comment (Josh-Cena) [inline website/src/data/users.tsx:103]:
```suggestion
    source: 
      'https://github.com/algolia/docsearch/tree/next/packages/website',
```

You would prefer `next`, right?

Comment (shortcuts) [reply website/src/data/users.tsx:103]: It's already the default branch and I'll do the renames soon so I guess we can keep the scope large to the repo, wdyt?

Comment (shortcuts) [reply website/src/data/users.tsx:103]: Ok make sense! I'll do it tomorrow and ping you here once done :D

Comment (Josh-Cena) [reply website/src/data/users.tsx:103]: Great, thanks!

Comment (shortcuts) [reply website/src/data/users.tsx:103]:
```suggestion
    source:
      'https://github.com/algolia/docsearch/tree/main/packages/website',
```

Comment (shortcuts) [reply website/src/data/users.tsx]: All good!

**Review by Josh-Cena — APPROVED** [review]
Thank you!

**Pull request merged**